### ISSUE TITLE
zsh-specific syntax breaks wait loop on null_resource.update_config_map_aws_auth

### ISSUE THREAD

**Issue by marcelloromani** [terraform-aws-modules/terraform-aws-eks#238]
# I have issues

## I'm submitting a...

* [x] bug report
* [ ] feature request
* [ ] support request
* [ ] kudos, thank you, warm fuzzy

## What is the current behavior?

When provisioning a new cluster on a system which doesn't have zsh, but has e.g. bash, the line

```
for i in {1..5}; do [...] ; done
```

causes the loop body to be executed only once, because `{1..5}` is interpreted as a single token.

This means that the `kubectl` command in the loop body is tried only once, which defeats the whole purpose of the original pull request.

## If this is a bug, how to reproduce? Please include a code sample if relevant.

Execute the module on a system which uses bash.

Redirect the cluster URL to localhost via `/etc/hosts`, so that regardless of the actual status of the cluster, the `kubectl apply` command always fails.

Adding a debug statement in the for loop to print the value of `$i` shows that instead of running for up to five times with values `1 2 3 4 5`, the `i` variable assumes only one value, namely `{1..5}`

Taint the resource, in order to force the for loop to run: `terraform taint -module=eks null_resource.update_config_map_aws_auth`

Repeat terraform apply

## What's the expected behavior?

If the `kubectl apply` command within the for loop fails, it should run up to 5 times before terminating.

## Are you able to fix this problem and submit a PR? Link here if you have already.

https://github.com/terraform-aws-modules/terraform-aws-eks/pull/245

## Environment details

* Affected module version: 1.8.0
* OS: Mac Os 10.13.6
* Terraform version: 0.11.11

## Any other relevant info
The code change was introduced in this commit:

https://github.com/terraform-aws-modules/terraform-aws-eks/commit/81706207fc8bac5b8c4649e8468617127f2a83f1

Another issue with the aforementioned code is that the null_resource always succeeds even if the `kubectl apply` command fails. The shell line should fail instead (e.g. `exit 1`) if the `kubectl apply` command fails.

**Comment by RothAndrew** [comment]
I had issues with this as well, I submitted (and got merged) the `local_exec_interpreter` variable, that should let you specify whatever interpreter you want. I use `local_exec_interpreter = ["bash", "-c"]`. The default is carried over from the default in Terraform, which is `[ "/bin/sh", "-c" ]`

**Comment by marcelloromani** [comment]
@RothAndrew in a way this could be viewed as an issue with the interpreter.

The reason I think the problem lies in the command instead is because it uses a `zsh`-specific (as fare as I could tell) syntax, which silently fails in `bash`, whereas something more traditional like `seq 1 5` works equally well in `zsh` and `bash`.

**Comment by marcelloromani** [comment]
I believe these are related issues:

#199 

#231

**Comment by marcelloromani** [comment]
This issue was fixed a while ago, so closing.
